Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/EventBased/PassportAnalytics.cs

```diff
@@ -0,0 +1,62 @@
+﻿using DCL.Passport;
+using Segment.Serialization;
+using System;
+
+namespace DCL.PerformanceAndDiagnostics.Analytics
+{
+    public class PassportAnalytics : IDisposable
+    {
+        private readonly IAnalyticsController analytics;
+        private readonly PassportController passportController;
+
+        public PassportAnalytics(IAnalyticsController analytics, PassportController passportController)
+        {
+            this.analytics = analytics;
+            this.passportController = passportController;
+
+            this.passportController.PassportOpened += OnPassportOpened;
+            this.passportController.BadgesSectionOpened += OnBadgesSectionOpened;
+            this.passportController.BadgeSelected += OnBadgeSelected;
+        }
+
+        public void Dispose()
+        {
+            passportController.PassportOpened -= OnPassportOpened;
+            passportController.BadgesSectionOpened -= OnBadgesSectionOpened;
+            this.passportController.BadgeSelected -= OnBadgeSelected;
+        }
+
+        private void OnBadgeSelected(string id, bool isOwnPassport)
+        {
+            analytics.Track(AnalyticsEvents.Profile.BADGE_UI_CLICK, new JsonObject
+            {
+                { "badge_id", id }, {"own_passport", isOwnPassport}
+            });
+        }
+
+        private void OnBadgesSectionOpened(string userId, bool isOwnPassport, string origin)
+        {
+            analytics.Track(AnalyticsEvents.Profile.BADGES_TAB_OPENED, new JsonObject
+            {
+                { "receiver_id", userId }, {"own_passport", isOwnPassport}, {"origin", origin}
+            });
+        }
+
+        private void OnPassportOpened(string userId, bool isOwnPassport)
+        {
+            if (isOwnPassport)
+            {
+                analytics.Track(AnalyticsEvents.Profile.OWN_PROFILE_OPENED, new JsonObject
+                {
+                    { "receiver_id", userId },
+                });
+                return;
+            }
+
+            analytics.Track(AnalyticsEvents.Profile.PASSPORT_OPENED, new JsonObject
+            {
+                { "receiver_id", userId }
+            });
```

Comment:
you can simplify it by presenting 
`string event = isOwnPassposrt? (AnalyticsEvents.Profile.OWN_PROFILE_OPENED, AnalyticsEvents.Profile.PASSPORT_OPENED);`
and then using it in just one event

Reply:
goood idea!